source/containers.tex

```diff
@@ -700,6 +700,31 @@
 
 \end{libreqtab4a}
 
+\pnum
+In this Clause the behavior of certain member functions and deduction guides
+depends on whether types qualify as input iterators or allocators.
+The extent to which an implementation determines that a type cannot be an input
+iterator is unspecified, except that as a minimum integral types shall not qualify
+as input iterators.
+Likewise, the extent to which an implementation determines that a type cannot be
+an allocator is unspecified, except that as a minimum a type \tcode{A} shall not qualify
+as an allocator unless it satisfies both of the following conditions:
+
+\begin{itemize}
+\item The \grammarterm{qualified-id} \tcode{A::value_type}
+is valid and denotes a type~(\ref{temp.deduct}).
+
+\item The expression \tcode{declval<A\&>().allocate(size_t\{\})}
+is well-formed when treated as an unevaluated operand.
+\end{itemize}
+
+\pnum
+\begin{note}
```

Comment:
I don't like this note as-written: it suggests that implementations can impose unbounded constraints on the types in questions, rather than merely constraints that are derivable from the allocator or input iterator requirements.

I'm inclined to merge this without the new note.